frontend/src/app/features/dashboard/task-form/task-form.component.ts

```diff
@@ -0,0 +1,104 @@
+import * as dashboardActions from '@store/actions/dashboard.actions';
+
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { endRepeatingValidator, getDescriptionValidators, getTitleValidators, repeatingPeriodValidator, startBeforeEndValidator } from '@utils/validators/task-validators';
+import { getEnd, getPeriodLabel, getStart } from '@utils/form-helpers/task-form.helper';
+
+import { BaseComponent } from '@shared/components/base/base.component';
+import { NewTask } from '#types/dashboard/tasks/new-task';
+import { Store } from '@ngrx/store';
+import { TaskRepeatingPeriod } from '#types/dashboard/tasks/enums/task-repeating-period';
+import { getRepeatingPeriods } from '@utils/creation-functions/repeating-period.helper';
+import { hasLengthErrors } from '@utils/validators/common-validators';
+import { selectChosenDashboardId } from '@store/selectors/dashboard.selectors';
+
+@Component({
+    selector: 'so-task-form',
+    templateUrl: './task-form.component.html',
+    styleUrls: ['./task-form.component.sass']
+})
+export class TaskFormComponent extends BaseComponent implements OnInit {
+    public taskForm: FormGroup = {} as FormGroup;
+
+    public isRepeating: boolean = false;
+    public selectStartTime: boolean = false;
+    public selectEndTime: boolean = false;
+
+    public repeatingOptions: TaskRepeatingPeriod[] = getRepeatingPeriods();
+
+    constructor(private store: Store) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.taskForm = this.buildTaskFrom();
+    }
+
+    public getLabel(period: TaskRepeatingPeriod): string {
+        return getPeriodLabel(period);
+    }
+
+    public validate(): void {
+        this.taskForm.get('startDate')?.updateValueAndValidity();
+        this.taskForm.get('endDate')?.updateValueAndValidity();
+        this.taskForm.get('endRepeating')?.updateValueAndValidity();
+        this.update();
+    }
+
+    public update(): void {
+        this.isRepeating = (this.taskForm.get('repeatingPeriod')?.value as TaskRepeatingPeriod) !== TaskRepeatingPeriod.None;
+        this.selectStartTime = this.taskForm.get('startTimeCheckbox')?.value;
+        this.selectEndTime = this.taskForm.get('endTimeCheckbox')?.value;
+    }
+
+    public hasLengthErrors(controlName: string): boolean {
+        return hasLengthErrors(this.taskForm, controlName);
+    }
+
+    public createTask(): void {
+        this.store.select(selectChosenDashboardId).pipe(
+            this.untilDestroyed
```

Comment:
i guess there should be take(1) to avoid any possible issues if for some reasons component won't be destroyed in future (e.g. it will be used somewhere with another lifetime).

Reply:
Fixed